Breadth_First_Search/BFS_in_Csharp.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Collections.Generic;                        // for generic data structures  
+
+namespace GraphBfs
+{
+    class graph
+    {
+       private int vertices;                           // no of vertices in a graph
+       private List<Int32>[] adj;                      //declaraing list
+       public graph(int v)                    
+       {
+           vertices=v;
+           adj=new List<Int32>[v];                     //initialising Adjacency List with number of vertices
+           for(int i=0;i<v;i++)
+           {
+               adj[i]=new List<int>();                 //initialising adjacency list for each node of graph
+
+           }
+       }
+       public void addedge(int c,int v)                  
+       {                                                 
+           adj[c].Add(v);                                
+           adj[v].Add(c);                                
+       }                                                  
+       public void BFS(int v)                          // function for Breadth First Search ,starting node as parameter
+       {
+           bool[] visited=new bool[vertices];          // visited array 
+           Queue<Int32> queue =new Queue<int>();       // queue declared
+           visited[v]=true;                            
+           queue.Enqueue(v);                           // enqueuing first node in queue
+           while(queue.Count!=0)
+           {
+               v=queue.Dequeue();                      // dequeue node from beginning of queue
+               Console.Write(v + " ");
+               foreach(int i in adj[v])
+               {
+                   if(!visited[i])                    //only unvisited node pushed into queue
+                   {
+                       visited[i]=true;               //marked visited
+                       queue.Enqueue(i);              //pushed into queue
+                   }
+               }
+           }
+       }
+
+    }
+    class program                                   
+    {
+        static void Main(string[] args)
+        {   
+            Console.WriteLine("Enter number of vertices in a graph");     //Taking input number of vertices
+            int v = Convert.ToInt32(Console.ReadLine());    
+
+            graph g=new graph(v);  
+
+            Console.WriteLine("Enter number of edge in a graph");        //Taking input number of edges
+            int e = Convert.ToInt32(Console.ReadLine());  
+
+            for(int i=0;i<e;i++)                                         
+            {   
+                Console.WriteLine("Enter nodes of " + (i+1) +" edge");     //taking input of edges
```

Comment:
Please change the place of comment, add it before the code, not on the sides across the file. Also please add spaces around operators across the file.

Reply:
@singh-shreya6 Position of comments changed